ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -605,8 +607,10 @@ func (r *RayClusterReconciler) buildWorkerPod(instance rayiov1alpha1.RayCluster,
 	podName := strings.ToLower(instance.Name + common.DashSymbol + string(rayiov1alpha1.WorkerNode) + common.DashSymbol + worker.GroupName + common.DashSymbol)
 	podName = utils.CheckName(podName) // making sure the name is valid
 	svcName := utils.GenerateServiceName(instance.Name)
-	podTemplateSpec := common.DefaultWorkerPodTemplate(instance, worker, podName, svcName)
-	pod := common.BuildPod(podTemplateSpec, rayiov1alpha1.WorkerNode, worker.RayStartParams, svcName, instance.Spec.EnableInTreeAutoscaling)
+	headPort := common.GetHeadPort(instance.Spec.HeadGroupSpec.RayStartParams)
```

Comment:
We could try to refactor to deduplicate the buildWorkerPod and and buildHeadPod logic.

The fact that HeadGroupSpec and WorkerGroupSpec are different types makes that a bit awkward, though.

Reply:
Haha, once we switch to Go 1.18 we could possibly use generics for this :)